### PR TITLE
gamepad-tool: init at 1.2

### PR DESCRIPTION
**Pull request by gador**
###### Description of changes

Gamepad-tool is an utility to map gampepad or joystick buttons and axis to a controller. Most useful for `moonlight` and gaming with it.  If one follows the [https://github.com/moonlight-stream/moonlight-docs/wiki/Gamepad-Mapping](official instructions of moonlight) you'll need this tool.

Although it hasn't been updated since 2018 it still works just fine.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 235580` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gamepadtool</li>
</ul>
</details>

File: pkgs/games/gamepadtool/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchurl, dpkg, lib, qt5, autoPatchelfHook, SDL2 }:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/games/gamepadtool/default.nix:3]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

**Review by drupol — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
@GrahamcOfBorg build gamepadtool

**Review by drupol — CHANGES_REQUESTED** [review]
The commit log message needs to be rewritten.

The derivation is `gamepadtool` and not `gamepad-tool`, unless this is the other way around, but then in that case, please rename the PR accordingly.

**Comment by gador** [comment]
Hi @drupol thanks for the review.
I included your suggestions. Please have another look

File: pkgs/games/gamepad-tool/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{ stdenv, fetchurl, dpkg, lib, qt5, autoPatchelfHook, SDL2 }:
```
Comment (drupol) [inline pkgs/games/gamepad-tool/default.nix:1]: How about using `stdenvNoCC` ?

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/games/gamepad-tool/default.nix (hunk 2 above)
Comment (gador) [reply pkgs/games/gamepad-tool/default.nix]: good idea. Changed :+1:

**Review by drupol — APPROVED** [review]
LGTM, let's wait the end of the eval

**Review by drupol — APPROVED** [review]
